addon/components/base/bs-button.js

```diff
@@ -232,12 +244,59 @@ export default Component.extend(TypeClass, SizeClass, {
    * `<state>Text` property.
    * This property will automatically be set when using a click action that supplies the callback with an promise
    *
-   * @property textState
+   * @property state
    * @type String
    * @default 'default'
    * @protected
    */
-  textState: 'default',
+  state: 'default',
+
+  /**
+   * @property textState
+   * @alias state
+   * @deprecated
+   * @protected
+   */
+  textState: deprecatingAlias('state', {
+    id: 'ember-bootstrap.bs-button-text-state',
+    until: '3.0.0'
```

Comment:
Not sure if we really need to do this, as the prop was never declared public. Not sure if the `protected` label makes so much sense, I usually only declare props either `public` or `private`, but not here, probably because the code base is quite old...

But it down not do any harm (the deprecating alias), so let's keep it. But for the future, can you label alls the new props `@private` please?

Reply:
I wasn't quite sure how we notice in API docs that a property is yielded. This doesn't seem to be covered by JSDoc. But sure will change them to `@private`.

Since passing `textState` in was covered by integration tests, I thought this might be considered public. Therefore I added the deprecation. If it's not, I could just change the until value to a minor release (e.g. `2.2.0`) as ember does it for highly used private APIs?

Reply:
> This doesn't seem to be covered by JSDoc.

Yes, indeed. https://github.com/ember-learn/ember-cli-addon-docs has support for this, which I am quite jealous for ;)

> I could just change the until value to a minor release (e.g. 2.2.0) as ember does it for highly used private APIs?

Yes, but honestly I don't expect this to be used at all, so I am fine with just renaming and marking it as private actually!